CriticalMass/APIRequest.swift

```diff
@@ -1,22 +1,39 @@
 import Foundation
 
+enum APIRequestBuildError: Error {
+    case invalidURL
+}
+
 public protocol APIRequestDefining {
     associatedtype ResponseDataType: Decodable
     var endpoint: Endpoint { get }
     var httpMethod: HTTPMethod { get }
     var headers: HTTPHeaders? { get }
     var requiresBackgroundTask: Bool { get }
-    func makeRequest() -> URLRequest
+    func makeRequest() throws -> URLRequest
     func parseResponse(data: Data) throws -> ResponseDataType
+    func getQueryItems() -> [URLQueryItem]?
```

Comment:
Why is GetQueryItems a function and not a variable?

Reply:
no reason. could be a property as well

Reply:
I updated it to a property